### PR TITLE
[875] Improve the support of libraries in the explorer

### PR DESCRIPTION
**Pull request by gdaniel**
Bug: https://github.com/eclipse-syson/syson/issues/875

# PLEASE READ ALL ITEMS AND CHECK ONLY RELEVANT CHECKBOXES BELOW

## Project management

- [x] Has the pull request been added to the relevant milestone?
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `type:`)
- [x] Have the relevant issues been added to the same project milestone as the pull request?

## Changelog and release notes

- [x] Has the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [x] In case of a change with a visual impact, are there any screenshots in the `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [ ] In case of a key change, has the change been added to `Key highlights` section in `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?

## Documentation

- [x] Have you included an update of the [documentation](https://doc.mbse-syson.org) in your pull request? Please ask yourself if an update (installation manual, user manual, developer manual...) is needed and add one accordingly.

## Tests

- [x] Is the code properly tested? Any pull request (fix, enhancement or new feature) should come with a test (or several). It could be unit tests, integration tests or cypress tests depending on the context. Only doc and releng pull request do not need for tests.

### CODE REVIEW

File: backend/views/syson-tree-explorer-view/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/ExplorerServices.java [hunk 1]
```diff
@@ -0,0 +1,255 @@
+/*******************************************************************************
```
Comment (gdaniel) [inline backend/views/syson-tree-explorer-view/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/ExplorerServices.java:1]: This class should be removed once https://github.com/eclipse-sirius/sirius-web/pull/4256 is merged

File: backend/views/syson-tree-explorer-view/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/api/IExplorerServices.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
```
Comment (gdaniel) [inline backend/views/syson-tree-explorer-view/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/api/IExplorerServices.java:1]: This class should be removed once https://github.com/eclipse-sirius/sirius-web/pull/4256 is merged

File: backend/views/syson-tree-explorer-view/src/main/java/org/eclipse/syson/tree/explorer/view/services/api/ISysONFilterService.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+package org.eclipse.syson.tree.explorer.view.services.api;
+
+import java.util.List;
+
+/**
+ * Services to apply filters on SysON explorer.
+ * 
+ * @author gdaniel
+ */
+public interface ISysONFilterService {
```
Comment (AxelRICHARD) [inline backend/views/syson-tree-explorer-view/src/main/java/org/eclipse/syson/tree/explorer/view/services/api/ISysONFilterService.java:22]: ISysONFilterService => ISysONExplorerFilterService

File: backend/views/syson-tree-explorer-view/src/main/java/org/eclipse/syson/tree/explorer/view/services/SysONFilterService.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+import org.eclipse.syson.tree.explorer.view.services.api.ISysONFilterService;
+import org.springframework.stereotype.Service;
+
+/**
+ * Services to apply filters on SysON explorer.
+ * 
+ * @author gdaniel
+ */
+@Service
+public class SysONFilterService implements ISysONFilterService {
```
Comment (AxelRICHARD) [inline backend/views/syson-tree-explorer-view/src/main/java/org/eclipse/syson/tree/explorer/view/services/SysONFilterService.java:33]: SysONExplorerFilterService

**Review by AxelRICHARD — COMMENTED** [review]
I'll handle to renaming my self and merge the PR after the build.

**Pull request merged**